adapter/adapter.go

```diff
@@ -191,6 +191,39 @@ func GetCommandEntry(ctx context.Context, bundleName, commandName string) (data.
 	return entries[0], nil
 }
 
+// GetCommandEntryByTrigger accepts a tokenized parameter slice and returns any
+// associated data.CommandEntry instances. If the number of matching
+// commands is > 1, an error is returned.
+func GetCommandEntryByTrigger(ctx context.Context, tokens []string) (data.CommandEntry, error) {
+	finders, err := allCommandEntryFinders()
+	if err != nil {
+		return data.CommandEntry{}, err
+	}
+
+	entries, err := findAllEntriesByTrigger(ctx, tokens, finders...)
+	if err != nil {
+		return data.CommandEntry{}, err
+	}
+
+	if len(entries) == 0 {
+		return data.CommandEntry{}, ErrNoSuchCommand
+	}
+
+	if len(entries) > 1 {
+		log.
+			WithField("requested", strings.Join(tokens, " ")).
+			WithField("bundle0", entries[0].Bundle.Name).
+			WithField("command0", entries[0].Command.Name).
+			WithField("bundle1", entries[1].Bundle.Name).
+			WithField("command1", entries[1].Command.Name).
+			Warn("Multiple commands found")
+
+		return data.CommandEntry{}, ErrMultipleCommands
```

Comment:
I was wondering about this. Should this be an error?

Reply:
Possibly not long term, but how would we determine which command to choose?

Reply:
In the long term we may want to actually execute _all_ matching triggers. Especially if we have, for example, one or more "silent" triggers that exist just to collect data.

Reply:
Good point! Created https://github.com/getgort/gort/issues/194